android/src/com/mapswithme/maps/downloader/MigrationController.java

```diff
@@ -149,7 +149,7 @@ void start(boolean keepOld)
     if (mState == State.PROGRESS)
       return;
 
-    Location loc = LocationHelper.INSTANCE.getLastLocation();
+    Location loc = LocationHelper.INSTANCE.getLastKnownLocation();
```

Comment:
Для этого можно просто дёргать системный 

LocationManager.getLastKnownLocation(LocationManager.PASSIVE_PROVIDER);

Вообще и здесь тоже можно упростить и позицию получать из него, не перебирая все провайдеры.



> On Apr 1, 2016, at 1:43 PM, Alex Zolotarev <notifications@github.com> wrote:
> 
> In android/src/com/mapswithme/maps/downloader/MigrationController.java <https://github.com/mapsme/omim/pull/2672#discussion_r58189614>:
> 
> > @@ -149,7 +149,7 @@ void start(boolean keepOld)
> >      if (mState == State.PROGRESS)
> >        return;
> >  
> > -    Location loc = LocationHelper.INSTANCE.getLastLocation();
> > +    Location loc = LocationHelper.INSTANCE.getLastKnownLocation();
> Конечно важно! Если тебе нужна хоть какая-то позиция, не зависимо от времени протухания, то ты никак ее сейчас не получишь. Или нужно по времени у позиции сделать switch и разную логику.
> … <x-msg://8/#>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> 
>  <https://github.com/mapsme/omim/pull/2672#discussion_r58189451> <https://github.com/mapsme/omim/pull/2672/files/cc92bbc4b790b789ab9d08f4d3d2825564b6054a#r58189451>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2672/files/cc92bbc4b790b789ab9d08f4d3d2825564b6054a#r58189614>